Ckzed zmetd

saurabh-newera opened the pull request:
config/ app info/ metrics integration with cloud

ckzed committed:
Add zmetd daemon

zmetd will send metrics data to ZedCloud.

ckzed committed:
Add protometrics

Protometrics is code written by Saurabh for encoding polled metrics
data.

saurabh-zededa committed:
reading network info file instead of executing exec

Commit:
Fixed compilation issues with zmetrics.go

Commit:
Merge branch 'ckzed-zmetd' of github.com:zededa/go-provision into ckzed-zmetd

saurabh-zededa committed:
added timer

saurabh-zededa committed:
added http request for sending metrics to zedcloud

srinibas-zededa committed:
device status

saurabh-zededa committed:
Added missing proto wrapper(zmsg). Fixes for sending device metrics and info to gilas.

saurabh-zededa committed:
added top level proto structure(zmsg) and its .pb.go file (for testing purpose)

saurabh-zededa committed:
Send Device info to gilas (fields don't have actual value). This is skeleton for device info.

srinibas-zededa committed:
merging metric/info into zmanager

srinibas-zededa committed:
metric in proto 3

srinibas-zededa committed:
remove zmsg wraper on device side communication

srinibas-zededa committed:
err handling

srinibas-zededa committed:
config handler

srinibas-zededa committed:
publish the config as separate app instance

srinibas-zededa committed:
config to app instance JSON

saurabh-zededa committed:
onboarding api changes

srinibas-zededa committed:
compilation fixes

saurabh-zededa committed:
proto changes in onboarding

saurabh-zededa committed:
Merge branch 'ckzed-zmetd' of https://github.com/zededa/go-provision into ckzed-zmetd

saurabh-zededa committed:
proto changes for onboarding

srinibas-zededa committed:
config file

srinibas-zededa committed:
Merge branch 'ckzed-zmetd' of https://github.com/zededa/go-provision into ckzed-zmetd

srinibas-zededa committed:
updated some filds in proto file for sync across dev and cloud

saurabh-zededa committed:
commented validation of content type for now.

saurabh-zededa committed:
content type fix

srinibas-zededa committed:
cfg/image download with proto/json

srinibas-zededa committed:
dev common prorto

saurabh-zededa committed:
commented code related to lookupParam

srinibas-zededa committed:
update app status to cloud

srinibas-zededa committed:
url-cfg.jaon

saurabh-zededa committed:
added https mechanism and url changes

srinibas-zededa committed:
dev id changes

srinibas-zededa committed:
urls moved to https

srinibas-zededa committed:
device name

srinibas-zededa committed:
header checks

srinibas-zededa committed:
error case

srinibas-zededa committed:
some fixes

srinibas-zededa committed:
crash fix

srinibas-zededa committed:
metric push delay is now 60 sec

srinibas-zededa committed:
crash fix

srinibas-zededa committed:
xen/overlay changes

srinibas-zededa committed:
memory size

Commit:
Merge branch 'master' into ckzed-zmetd

saurabh-zededa committed:
xen fixes

saurabh-zededa committed:
xen fixes

saurabh-zededa committed:
sample config for x_86

saurabh-zededa committed:
indentation changes

saurabh-zededa committed:
indentation changes

eriknordmark reviewed (changes requested):
Some things I don't understand, some things which I don't think can be removed. And some cleanup (delete code instead of commenting out.)

I think code should be removed from device-steps.sh to match the things you remove from the client.

I'll try to look more at the detailed new code tonight.

saurabh-zededa committed:
removed bootloader and memory hard coding

saurabh-zededa committed:
added comment for commented client code, added deviceCert in GET/POST, fixes in metrics and config handler.

saurabh-zededa committed:
device registration fixes

saurabh-zededa committed:
structured config file in README

srinibas-zededa committed:
multiple app instances

srinibas-zededa committed:
Merge branch 'ckzed-zmetd' of https://github.com/zededa/go-provision into ckzed-zmetd

srinibas-zededa committed:
typo

srinibas-zededa committed:
publish app status to cloud, while updating status file

Commit:
readme

srinibas-zededa committed:
update README file with  sandbox information

saurabh-zededa committed:
addede lookupParam

srinibas-zededa committed:
at config level

srinibas-zededa committed:
deprecated config proto for now

eriknordmark committed:
Made it works with prov1.zededa.net

Also tested that for a new device, after failing to talk to prov1, there is no json file etc produced nor copied by device-steps.sh

saurabh-zededa committed:
Merge branch 'master' into ckzed-zmetd
